syntax = "proto3";

package tbBasics;

import "math.proto";

message CreateActorMessage {
    uint32 id = 1;
    string entity = 2;
    string variant = 3;
    tbMath.Vec3Message position = 4;
    tbMath.Vec3Message rotation = 5;
}

message DeleteActorMessage {
    uint32 id = 1;
}

message SetCameraMessage {
    uint32 id = 1;
    string name = 2;
}

message MoveActorMessage {
    uint32 id = 1;
    tbMath.Vec3Message position = 2;
    tbMath.Mat3Message rotation = 3;
}

message AnimateActorMessage {
    uint32 id = 1;
    string animation = 2;
}

message ApplyForceMessage {
    uint32 id = 1;
    float magnitude = 2;
    tbMath.Vec3Message direction = 3;
}
